dozer/leak.py

```diff
@@ -174,13 +175,18 @@ def stop(self):
         self.running = False
 
     def index(self, req):
-        floor = req.GET.get('floor', 0)
+        floor = req.GET.get('floor', 0) or 0
+        filtertext = req.GET.get('filter', '')
+        filterre = re.compile(filtertext, re.IGNORECASE) if filtertext else None
```

Comment:
This might raise a `re.error` if the user-provided regexp is not valid (e.g. contains unbalanced parentheses).

OTOH int(floor) might also raise and there's no code to handle that error condition either, so I suppose it's acceptable to show an internal server error page on invalid user input to a debugging tool...

Reply:
I agree. In that case the error should be displayed so it can be understood, which I did in 8725719. Otherwise it needs a input validation which is a bit of an overkill for the use case.